BUILD.bazel

```diff
@@ -1065,6 +1065,8 @@ pyx_library(
     ]),
     copts = COPTS,
     deps = [
+        "//:core_worker_lib",
+        "//:ray_util",
```

Comment:
I think `streaming:streaming_lib` already has dependencies on `core_worker_lib` and `ray_util`.  Adding here is unnecessary.

Reply:
That's what I had assumed too, but I was getting linker errors without it.

Reply:
Can you paste error to here. It works before, it should work still.

Reply:
(23:01:31) ERROR: BUILD.bazel:1054:1: Couldn't build file python/ray/streaming/_streaming.so: Linking of rule '//:python/ray/streaming/_streaming.so' failed (Exit 1)
	lld-link: error: undefined symbol: public: __cdecl ray::RayLog::RayLog(char const *, int, enum ray::RayLogLevel)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\buffer.h:57
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: __cdecl ray::LocalMemoryBuffer::LocalMemoryBuffer(unsigned char *, unsigned __int64, bool))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\id.h:423
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: static class ray::ObjectID __cdecl ray::BaseID<class ray::ObjectID>::FromBinary(class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> const &))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:34
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:47
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:65
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:81
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:85
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:90
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:102
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:114
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced 125 more times

	lld-link: error: undefined symbol: public: virtual __cdecl ray::RayLog::~RayLog(void)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\buffer.h:57
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: __cdecl ray::LocalMemoryBuffer::LocalMemoryBuffer(unsigned char *, unsigned __int64, bool))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\buffer.h:57
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(int `public: __cdecl ray::LocalMemoryBuffer::LocalMemoryBuffer(unsigned char *, unsigned __int64, bool)'::`1'::dtor$11)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\id.h:423
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: static class ray::ObjectID __cdecl ray::BaseID<class ray::ObjectID>::FromBinary(class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> const &))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\id.h:423
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(int `public: static class BaseID<class ray::ObjectID>::ObjectID __cdecl ray::BaseID<class ray::ObjectID>::FromBinary(class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> const &)'::`1'::dtor$12)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:34
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:47
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:34
	>>>               streaming_lib.lib(data_reader.obj):(int `public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>>::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64)'::`1'::dtor$4)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:47
	>>>               streaming_lib.lib(data_reader.obj):(int `public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>>::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64)'::`1'::dtor$11)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:65
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:81
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced 260 more times

	lld-link: error: undefined symbol: public: class ray::ActorID const & __cdecl ray::WorkerContext::GetCurrentActorID(void) const
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\queue\queue_client.h:25
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: __cdecl ray::streaming::ReaderClient::ReaderClient(class ray::CoreWorker *, class ray::RayFunction &, class ray::RayFunction &))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\queue\queue_client.h:48
	>>>               bazel-out/x64_windows-fastbuild/bin/_objs/python/ray/streaming/_streaming.so/_streaming.obj:(public: __cdecl ray::streaming::WriterClient::WriterClient(class ray::CoreWorker *, class ray::RayFunction &, class ray::RayFunction &))

	lld-link: error: undefined symbol: public: static bool __cdecl ray::RayLog::IsLevelEnabled(enum ray::RayLogLevel)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:34
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:47
	>>>               streaming_lib.lib(data_reader.obj):(public: void __cdecl ray::streaming::DataReader::Init(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, class std::vector<class ray::ActorID, class std::allocator<class ray::ActorID>> const &, __int64))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:65
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:81
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:85
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannel(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:90
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:102
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:114
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::InitChannelMerger(void))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:122
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::GetMessageFromChannel(struct ray::streaming::ConsumerChannelInfo &, class std::shared_ptr<struct ray::streaming::DataBundle> &))
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\data_reader.cc:129
	>>>               streaming_lib.lib(data_reader.obj):(private: enum ray::streaming::StreamingStatus __cdecl ray::streaming::DataReader::GetMessageFromChannel(struct ray::streaming::ConsumerChannelInfo &, class std::shared_ptr<struct ray::streaming::DataBundle> &))
	>>> referenced 73 more times

	lld-link: error: undefined symbol: class std::basic_ostream<char, struct std::char_traits<char>> & __cdecl ray::operator<<(class std::basic_ostream<char, struct std::char_traits<char>> &, class ray::ObjectID const &)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\util\logging.h:71
	>>>               streaming_lib.lib(data_reader.obj):(public: class ray::RayLogBase & __cdecl ray::RayLogBase::operator<<<class ray::ObjectID>(class ray::ObjectID const &))
	>>> referenced by streaming_lib.lib(data_writer.obj)
	>>> referenced by streaming_queue.lib(queue_handler.obj)
	>>> referenced by streaming_lib.lib(channel.obj)

	lld-link: error: undefined symbol: unsigned __int64 __cdecl ray::MurmurHash64A(void const *, int, unsigned int)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\id.h:447
	>>>               streaming_lib.lib(data_reader.obj):(public: unsigned __int64 __cdecl ray::BaseID<class ray::ObjectID>::Hash(void) const)
	>>> referenced by streaming_lib.lib(data_writer.obj)
	>>> referenced by streaming_queue.lib(queue_handler.obj)
	>>> referenced by streaming_lib.lib(channel.obj)

	lld-link: error: undefined symbol: class std::basic_ostream<char, struct std::char_traits<char>> & __cdecl ray::operator<<(class std::basic_ostream<char, struct std::char_traits<char>> &, class ray::JobID const &)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\util\logging.h:71
	>>>               streaming_lib.lib(data_writer.obj):(public: class ray::RayLogBase & __cdecl ray::RayLogBase::operator<<<class ray::JobID>(class ray::JobID const &))

	lld-link: error: undefined symbol: class std::basic_ostream<char, struct std::char_traits<char>> & __cdecl ray::operator<<(class std::basic_ostream<char, struct std::char_traits<char>> &, class ray::ActorID const &)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\util\logging.h:71
	>>>               streaming_queue.lib(queue_handler.obj):(public: class ray::RayLogBase & __cdecl ray::RayLogBase::operator<<<class ray::ActorID>(class ray::ActorID const &))

	lld-link: error: undefined symbol: private: void __cdecl ray::Status::CopyFrom(struct ray::Status::State const *)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\status.h:224
	>>>               streaming_queue.lib(queue_handler.obj):(public: void __cdecl ray::Status::operator=(class ray::Status const &))
	>>> referenced by streaming_lib.lib(channel.obj)

	lld-link: error: undefined symbol: public: class ray::Status __cdecl ray::CoreWorker::SubmitActorTask(class ray::ActorID const &, class ray::RayFunction const &, class std::vector<class ray::TaskArg, class std::allocator<class ray::TaskArg>> const &, struct ray::TaskOptions const &, class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> *)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\queue\transport.cc:33
	>>>               streaming_queue.lib(transport.obj):(private: virtual void __cdecl ray::streaming::Transport::SendInternal(class std::shared_ptr<class ray::LocalMemoryBuffer>, class ray::RayFunction &, int, class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> &))

	lld-link: error: undefined symbol: public: class ray::Status __cdecl ray::CoreWorker::Get(class std::vector<class ray::ObjectID, class std::allocator<class ray::ObjectID>> const &, __int64, class std::vector<class std::shared_ptr<class ray::RayObject>, class std::allocator<class std::shared_ptr<class ray::RayObject>>> *)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\queue\transport.cc:53
	>>>               streaming_queue.lib(transport.obj):(public: virtual class std::shared_ptr<class ray::LocalMemoryBuffer> __cdecl ray::streaming::Transport::SendForResult(class std::shared_ptr<class ray::LocalMemoryBuffer>, class ray::RayFunction &, __int64))

	lld-link: error: undefined symbol: public: bool __cdecl ray::RayObject::IsException(enum ray::rpc::ErrorType *) const
	>>> referenced by (...)\execroot\com_github_ray_project_ray\streaming\src\queue\transport.cc:59
	>>>               streaming_queue.lib(transport.obj):(public: virtual class std::shared_ptr<class ray::LocalMemoryBuffer> __cdecl ray::streaming::Transport::SendForResult(class std::shared_ptr<class ray::LocalMemoryBuffer>, class ray::RayFunction &, __int64))

	lld-link: error: undefined symbol: public: class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> __cdecl ray::Status::ToString(void) const
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\status.h:213
	>>>               streaming_queue.lib(transport.obj):(class std::basic_ostream<char, struct std::char_traits<char>> & __cdecl ray::operator<<(class std::basic_ostream<char, struct std::char_traits<char>> &, class ray::Status const &))

	lld-link: error: undefined symbol: public: __cdecl ray::Status::Status(enum ray::StatusCode, class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> const &)
	>>> referenced by (...)\execroot\com_github_ray_project_ray\src\ray\common\status.h:110
	>>>               streaming_queue.lib(queue.obj):(public: static class ray::Status __cdecl ray::Status::OutOfMemory(class std::basic_string<char, struct std::char_traits<char>, class std::allocator<char>> const &))